corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -137,6 +137,21 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         return curr;
     }
 
+    function isStyleNPerRowRepeat(style) {
+        var pattern = /^\d+-per-row-repeat$/;
+        return pattern.test(style);
+    }
```

Comment:
could we add `per-row-repeats` as a constant similar we do for other appearance attributes? 

Could this actually reuse most of what's defined for `calculateElementWidth `?

Reply:
https://github.com/dimagi/commcare-hq/pull/33979/commits/5b6b1fe56b4bb555babdcadcf0bb168de3d27361

Reply:
Is there a way to also combine this into `calculateElementWidth`?

Reply:
Isn't `calculateElementWidth` providing a different return i.e. width of each item while this only returns a boolean if style is present?

Reply:
Cam also do something like get the element width back and check if its less than 12. If it is that means it is having that style applied and can use this condition to derive a boolean expression. Does this sound plausible?

Reply:
oh I see, my bad. I guess there isn't a good way to combine it without making it convoluted. Then just a nit: is this similar enough to how `getMatchingStyles` is used?

Reply:
It is already using `getMatcingStyles`

Probably can simplify like: `return getMatchingStyles(pattern, styleStr).length > 0`

Reply:
Ya basically the same as what's done for ` Question.prototype.stylesContains `

Reply:
https://github.com/dimagi/commcare-hq/pull/33979/commits/caa335077b3e082cc0e96b8f226d1fc84a483821